### ISSUE TITLE
PDP-10

### ISSUE THREAD

**Issue by larsbrinkhoff** [jeffpar/pcjs.v1#49]
Hello,

Which PDP-10 software are you planning to run?

I see there's a KA10 CPU in progress, but the latest version of TOPS-10 and TOPS-20 would run better on a KL10.  ITS will probably run on a KA10, but requires some special modification to a stock machine.

**Comment by jeffpar** [comment]
I'm starting with the oldest PDP-10 (KA10).  As for software, I'm going to start with PDP-10 diagnostics (eg, from paper tape or magnetic tape images).  However, I'm not yet to the point where I can run *any* off-the-shelf software for the PDP-10.  All I'm running so far are sequences that I hand-enter into the built-in debugger/assembler.

I haven't looked into what vintages of TOPS-10 and TOPS-20 are available yet, but there again, I would want to locate the oldest available version, start with that, then work forward to newer/newest versions.  In any case, I'm assuming that I'll need all the KA10 functionality, at a minimum.

I followed the same strategy with the PDP-11, by starting with the PDP-11/20 (more or less) and working my way up to the 11/70, and starting with diagnostic paper tape images and gradually working up to disk images with complete operating systems.

**Comment by larsbrinkhoff** [comment]
That sounds reasonable.  But be aware that the different PDP-10 models to some extent used different peripheral devices.  I would advice you to not invest too much effort emulating a KA10 system if your ultimate goal is to run later versions of the operating systems.

**Comment by larsbrinkhoff** [comment]
If you want some simple programs for testing, I can provide paper tape or magtape images with standalone tools that can read and write to tape and disk.

**Comment by jeffpar** [comment]
That would be great. Please share or point me to whatever you can. Thanks.

**Comment by larsbrinkhoff** [comment]
It's here: https://github.com/larsbrinkhoff/ka10-simh/commits/its

There's MAGDMP in bootable paper tape and magtape formats.  MAGDMP can read/write/run files from a appropriately formatted magtape.

There's also DSKDMP in paper tape format.  DSKDMP is like MAGDMP but for an ITS disk filesystem.

**Comment by larsbrinkhoff** [comment]
How's the PDP-10 emulator coming along?

**Comment by jeffpar** [comment]
Too many projects, too little time. This [PCjs page](https://www.pcjs.org/blog/2017/03/24/) demonstrates the state of that particular emulator so far. It's still just a CPU emulation at this point -- no device/interrupt support.  I also wrote a MACRO-10 work-alike assembler in JavaScript that's baked into the emulation environment, so that I can inject diagnostic source code into the machine without requiring binaries (or any means of loading them).  This [PCjs section](https://www.pcjs.org/apps/pdp10/diags/ka10/) of the website documents the diagnostics that I had worked through.
